include/utils.hpp

```diff
@@ -226,6 +228,51 @@ namespace utils {
     return date.utc_seconds;
   }
 
+  template <typename T>
+  void delete_table (const name & code, const name & scope) {
+
+    T table(code, scope.value);
+    auto itr = table.begin();
+
+    while (itr != table.end()) {
+      itr = table.erase(itr);
+    }
+
+  }
+
+  template <typename... T>
+  void send_deferred_transaction (
+    const name & code,
+    const permission_level & permission,
+    const name & contract, 
+    const name & action,  
+    const std::tuple<T...> & data) {
+
+    DEFINE_DEFERRED_ID_TABLE
```

Comment:
I understand how these are supposed to work, creating a new unique deferred ID for each action

But I wonder and maybe you know this - what's the scope of a deferred ID? Is it a contract scope? Or global EOSIO network scope? Or signer scope? Payer scope? 

Example if I send a deferred transaction from accts.seeds with id = 1
And some other contract sends a deferred action to accts.seeds from another contract also with id 1 - is that the same ID?

I looked that up at some point but maybe you know what these IDs are unique to.

Reply:
The scope of the deferred id is the ram payer, in seeds it happens to be always the contract. So in this case the ids would be `accts.seeds::1`, and the other contract `other::1`, so there is no conflict. 

There is no conflict in using the id=1 several times, the only condition is that a deferred transaction shouldn't be scheduled with that id. What I mean is that, if the transaction with id 1 is already executed, then that id is free again and can be reused.

I added that method to not worry about deferred ids, because I see that in the contracts the id is always the next.value or something like that and there is some repeated code to create the transaction. And if at some point deferred transactions are removed from eosio, we could change that single method to use some scheduler and automatically fix all the deferred code.